packages/host/app/components/operator-mode/card-adoption-chain.gts

```diff
@@ -0,0 +1,117 @@
+import Component from '@glimmer/component';
+//@ts-ignore cached not available yet in definitely typed
+import { cached } from '@glimmer/tracking';
+import { ModuleSyntax } from '@cardstack/runtime-common/module-syntax';
+import { loadCard } from '@cardstack/runtime-common/code-ref';
+import type { Ready } from '@cardstack/host/resources/file';
+import type { BaseDef } from 'https://cardstack.com/base/card-api';
+import CardSchemaEditor from '@cardstack/host/components/operator-mode/card-schema-editor';
+import LoaderService from '@cardstack/host/services/loader-service';
+import { service } from '@ember/service';
+import { getCardType } from '@cardstack/host/resources/card-type';
+import { tracked } from '@glimmer/tracking';
+import { type Type } from '@cardstack/host/resources/card-type';
+
+interface Signature {
+  Args: {
+    file: Ready;
+    importedModule: Record<string, any>;
+  };
+}
+
+export default class CardAdoptionChain extends Component<Signature> {
+  <template>
+    <style>
+      .card-adoption-chain {
+        height: 100%;
+        background-color: var(--boxel-200);
+        padding: var(--boxel-sp);
+        overflow-y: auto;
+      }
+    </style>
+
+    <div class='card-adoption-chain'>
+      <h3>Schema Editor</h3>
+
+      {{#each this.cardInheritanceChain as |data|}}
+        <CardSchemaEditor
+          @card={{data.card}}
+          @cardType={{data.cardType}}
+          @file={{@file}}
+          @moduleSyntax={{this.moduleSyntax}}
+        />
+      {{/each}}
+    </div>
+  </template>
+
+  @service declare loaderService: LoaderService;
+  @tracked cardInheritanceChain: {
+    cardType: Type;
+    card: any;
+  }[] = [];
+
+  constructor(owner: unknown, args: Signature['Args']) {
+    super(owner, args);
+    this.loadInheritanceChain();
```

Comment:
you should be performing an EX task not executing an async function. if this component is destroyed before async finishes errors will result.

Reply:
Good catch, I'm aware of this problem but forgot to switch to a task after prototyping the data loading. Thanks!